### PR TITLE
upgrades: add migration that creates system tables in all tenants

### PR DESCRIPTION
**Pull request by andy-kimball**
A change in 24.1 creates all system tables in new app tenants, even if the system tables are only used by the system tenant. However, app tenants created before 24.1 are missing these system tables. This commit adds a migration that adds the missing system tables to existing app tenants.

Epic: none

Release note: All system tables are now visible in app tenants that were created before v24.1.

### CODE REVIEW

**Review by renatolabs — APPROVED** [review]
:lgtm:

LGTM, but maybe @dt or @stevendanna should take a look as they're more familiar with this code.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/125979#-:-O-wC0Fq-aNIdg9SYPBW:b-xx5wk7)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @stevendanna)

File: pkg/upgrade/upgrades/v24_1_tenant_system_tables.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2024 The Cockroach Authors.
```
Comment (renatolabs) [inline pkg/upgrade/upgrades/v24_1_tenant_system_tables.go:1]: Shouldn't this file have a `v24_2` prefix?

Comment (andy-kimball) [reply pkg/upgrade/upgrades/v24_1_tenant_system_tables.go]: Yes, it should. Fixed.

File: pkg/upgrade/upgrades/v24_2_tenant_system_tables_test.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		checkTable("system.tenant_usage", shouldExist)
+		checkTable("system.span_configurations", shouldExist)
+		checkTable("system.task_payloads", shouldExist)
+		checkTable("system.tenant_tasks", shouldExist)
+		checkTable("system.tenant_id_seq", shouldExist)
+	}
+
+	checkTables(false /* shouldExist */)
+	upgrades.Upgrade(t, tenantDB, clusterversion.V24_2_TenantSystemTables, nil, false)
+	checkTables(true /* shouldExist */)
```
Comment (stevendanna) [inline pkg/upgrade/upgrades/v24_2_tenant_system_tables_test.go:115]: If you fancied, you could throw something like this:

```
+       // Check that the system and secondary tenant have all of the
+       // same tables in the system database.
+       tenantRunner := sqlutils.MakeSQLRunner(tenantDB)
+       systemTableNames := "SELECT name FROM crdb_internal.tables WHERE database_name ='system' ORDER BY name"
+       tenantRunner.CheckQueryResults(t,
+               systemTableNames,
+               sysDB.QueryStr(t, systemTableNames))
```

here as well. I tested it locally and that passed, so I think you've correctly gotten all of the missing tables.

**Review by stevendanna — APPROVED** [review]
Thanks a ton for cleaning this up!

File: pkg/upgrade/upgrades/v24_2_tenant_system_tables_test.go (hunk 2 above)
Comment (andy-kimball) [reply pkg/upgrade/upgrades/v24_2_tenant_system_tables_test.go:115]: Added.

**Pull request merged**